Dockerfile

```diff
@@ -0,0 +1,32 @@
+FROM registry.access.redhat.com/rhscl/ruby-23-rhel7
+
+# sshd
+USER root
+RUN yum install -y --setopt=tsflags=nodocs openssh-server libicu-devel && \
+    yum clean all && /usr/sbin/sshd-keygen
+
+RUN mkdir /var/run/sshd
+
+# TODO: is this needed? https://git.io/vrplM
+RUN sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd
+
+# gems
+RUN ["bash", "-c", "gem install --no-ri --no-rdoc bunny"]
+
+# git user
+RUN groupadd -r git &&\
+     useradd -r -s /bin/bash -g git git && \
+     mkdir --parent /home/git && \
+     chown -R git:git /home/git
+
+# gitlab-shell setup
+COPY . /home/git/gitlab-shell
+WORKDIR /home/git/gitlab-shell
+RUN ["bash", "-c", "./bin/install"]
+
+COPY authorized_keys /home/git/.ssh/authorized_keys
+
+USER root
```

Comment:
No matter what user you set in the Dockerfile, OS will run the container with some arbitrary user ID by default. If the image doesn't support that the deployment will fail with CrashLoopBackOff.

To allow running containers as root one needs to change the default security settings.

https://docs.openshift.com/enterprise/3.1/admin_guide/manage_scc.html#enable-images-to-run-with-user-in-the-dockerfile
https://docs.openshift.com/enterprise/3.1/admin_guide/manage_scc.html#enable-dockerhub-images-that-require-root